diff --git a/gdk/quartz/gdkselection-quartz.c b/gdk/quartz/gdkselection-quartz.c
index b3055324df..704c06e9bb 100644
--- a/gdk/quartz/gdkselection-quartz.c
+++ b/gdk/quartz/gdkselection-quartz.c
@@ -235,13 +235,13 @@ gdk_text_property_to_utf8_list_for_display (GdkDisplay    *display,
 GdkAtom
 gdk_quartz_pasteboard_type_to_atom_libgtk_only (NSString *type)
 {
-  if ([type isEqualToString:NSStringPboardType])
+  if ([type isEqualToString:NSPasteboardTypeString])
     return gdk_atom_intern_static_string ("UTF8_STRING");
-  else if ([type isEqualToString:NSTIFFPboardType])
+  else if ([type isEqualToString:NSPasteboardTypeTIFF])
     return gdk_atom_intern_static_string ("image/tiff");
-  else if ([type isEqualToString:NSColorPboardType])
+  else if ([type isEqualToString:NSPasteboardTypeColor])
     return gdk_atom_intern_static_string ("application/x-color");
-  else if ([type isEqualToString:NSURLPboardType])
+  else if ([type isEqualToString:NSPasteboardTypeURL])
     return gdk_atom_intern_static_string ("text/uri-list");
   else
     return gdk_atom_intern ([type UTF8String], FALSE);
@@ -251,13 +251,13 @@ NSString *
 gdk_quartz_target_to_pasteboard_type_libgtk_only (const char *target)
 {
   if (strcmp (target, "UTF8_STRING") == 0)
-    return NSStringPboardType;
+    return NSPasteboardTypeString;
   else if (strcmp (target, "image/tiff") == 0)
-    return NSTIFFPboardType;
+    return NSPasteboardTypeTIFF;
   else if (strcmp (target, "application/x-color") == 0)
-    return NSColorPboardType;
+    return NSPasteboardTypeColor;
   else if (strcmp (target, "text/uri-list") == 0)
-    return NSURLPboardType;
+    return NSPasteboardTypeURL;
   else
     return [NSString stringWithUTF8String:target];
 }
diff --git a/gtk/gtkquartz.c b/gtk/gtkquartz.c
index d630abc040..9499cbb8cd 100644
--- a/gtk/gtkquartz.c
+++ b/gtk/gtkquartz.c
@@ -140,7 +140,7 @@ _gtk_quartz_get_selection_data_from_pasteboard (NSPasteboard *pasteboard,
     selection_data->display = gdk_display_get_default ();
   if (target == gdk_atom_intern_static_string ("UTF8_STRING"))
     {
-      NSString *s = [pasteboard stringForType:NSStringPboardType];
+      NSString *s = [pasteboard stringForType:NSPasteboardTypeString];
 
       if (s)
 	{
@@ -191,7 +191,7 @@ _gtk_quartz_get_selection_data_from_pasteboard (NSPasteboard *pasteboard,
            gtk_selection_data_set_uris (selection_data, uris);
            g_free (uris);
          }
-      else if ([[pasteboard types] containsObject:NSURLPboardType])
+      else if ([[pasteboard types] containsObject:NSPasteboardTypeURL])
         {
           gchar *uris[2];
           NSURL *url = [NSURL URLFromPasteboard:pasteboard];
@@ -212,7 +212,7 @@ _gtk_quartz_get_selection_data_from_pasteboard (NSPasteboard *pasteboard,
       name = gdk_atom_name (target);
 
       if (strcmp (name, "image/tiff") == 0)
-	data = [pasteboard dataForType:NSTIFFPboardType];
+	data = [pasteboard dataForType:NSPasteboardTypeTIFF];
       else
 	data = [pasteboard dataForType:[NSString stringWithUTF8String:name]];
 
@@ -246,10 +246,10 @@ _gtk_quartz_set_selection_data_for_pasteboard (NSPasteboard     *pasteboard,
 
   type = gdk_quartz_atom_to_pasteboard_type_libgtk_only (gtk_selection_data_get_target (selection_data));
 
-  if ([type isEqualTo:NSStringPboardType]) 
+  if ([type isEqualTo:NSPasteboardTypeString])
     [pasteboard setString:[NSString stringWithUTF8String:(const char *)data]
                   forType:type];
-  else if ([type isEqualTo:NSColorPboardType])
+  else if ([type isEqualTo:NSPasteboardTypeColor])
     {
       guint16 *color = (guint16 *)data;
       float red, green, blue, alpha;
@@ -263,7 +263,7 @@ _gtk_quartz_set_selection_data_for_pasteboard (NSPasteboard     *pasteboard,
       nscolor = [NSColor colorWithDeviceRed:red green:green blue:blue alpha:alpha];
       [nscolor writeToPasteboard:pasteboard];
     }
-  else if ([type isEqualTo:NSURLPboardType])
+  else if ([type isEqualTo:NSPasteboardTypeURL])
     {
       gchar **list = NULL;
       int count;
