commit 96e0c97b316b7f05fe3da1d9a535f23de9e8a031
Author: Cody Russell <cody@jhu.edu>
Date:   Sun Jul 9 13:24:25 2017 -0500

    [Mac] Add additional OSX version numbers to Sierra

diff --git a/gdk/quartz/gdkquartz.h b/gdk/quartz/gdkquartz.h
index 558c53535..8677663ed 100644
--- a/gdk/quartz/gdkquartz.h
+++ b/gdk/quartz/gdkquartz.h
@@ -48,7 +48,11 @@ typedef enum
   GDK_OSX_SNOW_LEOPARD = 6,
   GDK_OSX_LION = 7,
   GDK_OSX_MOUNTAIN_LION = 8,
-  GDK_OSX_CURRENT = 8,
+  GDK_OSX_MAVERICKS = 9,
+  GDK_OSX_YOSEMITE = 10,
+  GDK_OSX_EL_CAPITAN = 11,
+  GDK_OSX_SIERRA = 12,
+  GDK_OSX_CURRENT = 12,
   GDK_OSX_NEW = 99
 } GdkOSXVersion;
 
