commit a8bd6caefcbe5462547a277a4fb864989626ef6f
Author: Cody Russell <cody@jhu.edu>
Date:   Mon Jan 27 10:44:06 2020 -0600

    Try to get the family name from sysfont

diff --git a/pango/pangocoretext-fontmap.c b/pango/pangocoretext-fontmap.c
index 30605fe..7124db4 100644
--- a/pango/pangocoretext-fontmap.c
+++ b/pango/pangocoretext-fontmap.c
@@ -1437,7 +1437,8 @@ pango_core_text_font_map_init (PangoCoreTextFontMap *ctfontmap)
   if (major == 10 && minor >= 11)
     {
       NSFont *sysfont = [NSFont systemFontOfSize: 0.0];
-      NSString *name = [[NSFont fontWithName: [[sysfont fontDescriptor] postscriptName] size: 0.0] familyName];
+      //NSString *name = [[NSFont fontWithName: [[sysfont fontDescriptor] postscriptName] size: 0.0] familyName];
+      NSString *name = [sysfont familyName];
 
       NSArray *fontfaces = [[NSFontManager sharedFontManager] availableMembersOfFontFamily: name];
       int num_faces = [fontfaces count];
