diff --git a/pango/.libs/libpango-1.0.0.dylib b/pango/.libs/libpango-1.0.0.dylib
index 61cf31d..ac8ae15 100755
Binary files a/pango/.libs/libpango-1.0.0.dylib and b/pango/.libs/libpango-1.0.0.dylib differ
diff --git a/pango/.libs/pango-context.o b/pango/.libs/pango-context.o
index 0fa9fe2..039e046 100644
Binary files a/pango/.libs/pango-context.o and b/pango/.libs/pango-context.o differ
diff --git a/pango/pango-context.c b/pango/pango-context.c
index 8e96713..d66f8bd 100644
--- a/pango/pango-context.c
+++ b/pango/pango-context.c
@@ -1459,7 +1459,8 @@ itemize_state_process_run (ItemizeState *state)
        */
       if (G_UNLIKELY (!g_unichar_isgraph (wc) &&
 		      g_unichar_type (wc) != G_UNICODE_PRIVATE_USE &&
-		      g_unichar_type (wc) != G_UNICODE_UNASSIGNED))
+		      g_unichar_type (wc) != G_UNICODE_UNASSIGNED) &&
+          wc != 0x3000)
 	{
 	  shape_engine = NULL;
 	  font = NULL;
diff --git a/tests/cxx-test b/tests/cxx-test
index 683868f..428d903 100755
Binary files a/tests/cxx-test and b/tests/cxx-test differ
