diff --git a/pango/pangocoretext-fontmap.c b/pango/pangocoretext-fontmap.c
old mode 100644
new mode 100755
index 27132c4..5de6209
--- a/pango/pangocoretext-fontmap.c
+++ b/pango/pangocoretext-fontmap.c
@@ -573,8 +573,8 @@ pango_core_text_family_list_faces (PangoFontFamily  *family,
       CFArrayRef ctfaces;
       CFArrayRef font_descriptors;
       CFDictionaryRef attributes;
-      CFIndex i, count;
-
+      CFIndex i;
+      CFIndex count = 0;
       CFTypeRef keys[] = {
           (CFTypeRef) kCTFontFamilyNameAttribute
       };
@@ -605,7 +605,9 @@ pango_core_text_family_list_faces (PangoFontFamily  *family,
 
       italic_faces = g_hash_table_new (g_direct_hash, g_direct_equal);
 
-      count = CFArrayGetCount (ctfaces);
+      if (ctfaces)
+        count = CFArrayGetCount (ctfaces);
+
       for (i = 0; i < count; i++)
         {
           PangoCoreTextFace *face;
